src/store.js

```diff
@@ -427,3 +427,155 @@ export const useWizardStore = defineStore("wizard", {
     },
   },
 });
+
+export const useMirrorsStore = defineStore("mirrors", {
+  state: () => ({
+    // URL lists
+    idf_urls: [],
+    tools_urls: [],
+    pypi_urls: [],
+
+    // Latency maps (url -> ms; 0 means timeout/unreachable; undefined means not yet measured)
+    idf_latency_map: {},
+    tools_latency_map: {},
+    pypi_latency_map: {},
+
+    // Selected (from backend quick URL endpoints)
+    selected_idf: "",
+    selected_tools: "",
+    selected_pypi: "",
+
+    // Loading flags
+    loading_idf_urls: false,
+    loading_tools_urls: false,
+    loading_pypi_urls: false,
+    loading_idf_latency: false,
+    loading_tools_latency: false,
+    loading_pypi_latency: false,
+
+    // Last updated timestamps (ms epoch)
+    idf_last_updated: 0,
+    tools_last_updated: 0,
+    pypi_last_updated: 0,
+
+    // TTL for latency cache (15 minutes)
+    latency_ttl_ms: 15 * 60 * 1000,
+  }),
+  getters: {
+    idfUrls: (state) => state.idf_urls,
+    toolsUrls: (state) => state.tools_urls,
+    pypiUrls: (state) => state.pypi_urls,
+    idfLatencyMap: (state) => state.idf_latency_map,
+    toolsLatencyMap: (state) => state.tools_latency_map,
+    pypiLatencyMap: (state) => state.pypi_latency_map,
+  },
+  actions: {
+    // Backend uses Option<u32> for latency values; Timedout values are represented as None. 
+    // We normalize to 0 for timeout and the value for the latency. If the value is undefined, we return undefined as it means the mirror is not yet measured.
+    normalizeLatencyValue(value) {
```

Comment:
Since Rust sends Option<u32>, Tauri serialises None as null and Some(x) as x. So just use:

number → measured latency
null → timeout
(keep undefined as a unmeasured state)
no need for normalizeLatencyValue at all